code/modules/urist/mob/uristanimals.dm

```diff
@@ -164,4 +164,172 @@ Please keep it tidy, by which I mean put comments describing the item before the
+	icon_state = "goose_labcoat"
+	icon_living = "goose_labcoat"
+	icon_dead = "goose_labcoat_dead"
+
+
+//POSSUM!
+/mob/living/simple_animal/opossum
+	name = "opossum"
+	real_name = "opossum"
+	desc = "It's an opossum, a small scavenging marsupial."
+	icon_state = "possum"
+	item_state = "possum"
+	icon_living = "possum"
+	icon_dead = "possum_dead"
+	icon = 'icons/uristmob/possum.dmi'
+	speak_emote = list("hisses")
+	speak = list("Hiss!","Aaa!","Aaa?")
+	emote_hear = list("hisses")
+	emote_see = list("forages for trash", "lounges")
+	speak_chance = 1
+	pass_flags = PASS_FLAG_TABLE
+	turns_per_move = 3
+	see_in_dark = 6
+	maxHealth = 50
+	health = 50
+	response_help = "pets"
+	response_disarm = "gently pushes aside"
+	response_harm = "stamps on"
+	density = FALSE
+	minbodytemp = 223
+	maxbodytemp = 323
+	universal_speak = FALSE
+	universal_understand = TRUE
+	holder_type = /obj/item/weapon/holder/possum
+	mob_size = MOB_SMALL
+	possession_candidate = 1
+	can_escape = TRUE
+	can_pull_size = ITEM_SIZE_SMALL
+	can_pull_mobs = MOB_PULL_SMALLER
+	var/is_angry = FALSE
+
+
+
+
+/mob/living/simple_animal/opossum/adjustBruteLoss(damage)
+	. = ..()
+	if(damage >= 3)
+		respond_to_damage()
+
+
+
+
+
+/mob/living/simple_animal/opossum/proc/respond_to_damage()
+	if(!resting && stat == CONSCIOUS)
+		if(!is_angry)
+			is_angry = TRUE
+			custom_emote(src, "hisses!")
+		else
+			resting = TRUE
+			custom_emote(src, "dies!")
+		update_icon()
+
+/mob/living/simple_animal/opossum/on_update_icon()
+
+	if(stat == DEAD || (resting && is_angry))
+		icon_state = icon_dead
+	else if(resting || stat == UNCONSCIOUS)
+		icon_state = "[icon_living]_sleep"
+	else if(is_angry)
+		icon_state = "[icon_living]_aaa"
+	else
+		icon_state = icon_living
+
+/mob/living/simple_animal/opossum/Initialize()
+	. = ..()
+	verbs += /mob/living/proc/ventcrawl
+	verbs += /mob/living/proc/hide
+
+/mob/living/simple_animal/opossum/poppy
+	name = "Poppy the Safety Possum"
+	desc = "It's an opossum, a small scavenging marsupial. It's wearing appropriate personal protective equipment, though."
+	icon_state = "poppy"
+	item_state = "poppy"
+	icon_living = "poppy"
+	icon_dead = "poppy_dead"
+	holder_type = /obj/item/weapon/holder/possum/poppy
+	var/aaa_words = list("delaminat", "meteor", "fire", "breach")
+
+/mob/living/simple_animal/opossum/poppy/hear_broadcast(datum/language/language, mob/speaker, speaker_name, message)
+  . = ..()
+  check_keywords(message)
+
+/mob/living/simple_animal/opossum/poppy/hear_say(message, verb = "says", datum/language/language = null, alt_name = "",italics = 0, mob/speaker = null, sound/speech_sound, sound_vol)
+  . = ..()
+  check_keywords(message)
+
+/mob/living/simple_animal/opossum/poppy/proc/check_keywords(message)
+	if(!client && stat == CONSCIOUS)
+		message = lowertext(message)
+		for(var/aaa in aaa_words)
+			if(findtext(message, aaa))
+				respond_to_damage()
+				return
+
```

Comment:
I know it's Baycode, but this would be a cleaner factorization:
```suggestion
/mob/living/simple_animal/opossum/poppy/hear_say(message, verb = "says", datum/language/language = null, alt_name = "",italics = 0, mob/speaker = null, sound/speech_sound, sound_vol)
  . = ..()
  var/keywords_found = check_keywords(message)
  if(keywords_found)
      respond_to_damage()

/mob/living/simple_animal/opossum/poppy/proc/check_keywords(message)
	if(!client && stat == CONSCIOUS)
		message = lowertext(message)
		
		for(var/aaa in aaa_words)
			if(findtext(message, aaa))
				return TRUE
	
	return FALSE

```